### ISSUE TITLE
Webpack 5 ReferenceError: process is not defined

### ISSUE THREAD

**Issue by quanbrew** [pmndrs/jotai#225]
Since webpack 5 remove `process`.

```
Uncaught ReferenceError: process is not defined
    at eval (index.js?6d4b:110)
    at ./node_modules/jotai/index.js (main.877e82636910ffdd80a5.js:784)
    at Module.options.factory (main.7695fd2e2d6cdbc8ed8a.js:1180)
    at __webpack_require__ (main.7695fd2e2d6cdbc8ed8a.js:627)
    at fn (main.7695fd2e2d6cdbc8ed8a.js:867)
    at eval (App.tsx:7)
    at ./src/App.tsx (main.7695fd2e2d6cdbc8ed8a.js:24)
    at Module.options.factory (main.7695fd2e2d6cdbc8ed8a.js:1180)
    at __webpack_require__ (main.7695fd2e2d6cdbc8ed8a.js:627)
    at fn (main.7695fd2e2d6cdbc8ed8a.js:867)
```

Cause by this line: 
```
const isReactExperimental = !!process.env.IS_REACT_EXPERIMENTAL || !!React.unstable_useMutableSource;
```

**Comment by Aslemammad** [comment]
@quanbrew Could you reproduce the problem, codesandbox or github source?

**Comment by dai-shi** [comment]
Does this help?

```diff
diff --git a/src/core/Provider.ts b/src/core/Provider.ts
index 1686b68..9e8e91a 100644
--- a/src/core/Provider.ts
+++ b/src/core/Provider.ts
@@ -40,7 +40,7 @@ import { AtomState, State, getContexts } from './contexts'
 
 // guessing if it's react experimental channel
 const isReactExperimental =
-  !!process.env.IS_REACT_EXPERIMENTAL ||
+  !!(typeof process === 'object' && process.env.IS_REACT_EXPERIMENTAL) ||
   !!(React as any).unstable_useMutableSource
 
 const useWeakMapRef = <T extends WeakMap<object, unknown>>() => {
```

**Comment by Aslemammad** [comment]
@dai-shi I'm gonna test it and make PR about.

**Comment by Aslemammad** [comment]
Fixed via #226
